### PR TITLE
VideoPress: Allow user to update poster selecting a frame from video

### LINKED ISSUE #26153: VideoPress: Implement updating poster image from video frame selector
- [ ] Extract the video frame selector from the block to be reused on the Edit Details page - https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-uploader/uploader-editor.js#L31
- [ ] Extract the hook used to update the poster image
- [ ] Implementing it at `useEditDetails` hook

### PR DESCRIPTION
**Pull request by renatoagds**
Implement the feature to allow the user select a poster image using a frame from the video.

Closes #26153 

### ⚠️ KNOWN BUG

If you reload the page, it won't use the new poster image, for some reason we're not getting fresh data. But if you navigate to library and open the attachment, the player will load based on fresh data.

#### Changes proposed in this Pull Request:

* Move `usePosterUpload` and `usePosterImage` hooks out of `videopress block`.
* Extract `VideoFrameSelector` from `videopress block`.
* Fix dropdown not closing in `VideoThumbnail`.
* Allow rendering node in `thumbnail` space at `VideoThumbnail`.
* Support videopress shared components in `Storybook`.
* Extract `PlayIcon` from `videopress block`.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

N/A

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Start `VideoPress`
* Navigate to its` dashboard `.
* Click on `EditDetails` button.
* At `thumbnail`, click on `edit` button.
* Select the second option - `Select from `video`
* Use the range to select a frame.
* Click on `Select Frame`.
* And save changes.
* You should wait and see the new poster from frame.
* Confirm it from dev tools.

#### Demo

https://user-images.githubusercontent.com/1663717/191580266-4a8cfd43-9c44-414d-b17b-84c492343aa6.mp4

### CODE REVIEW

File: projects/packages/videopress/src/client/admin/components/edit-video-details/use-edit-details.ts [hunk 1]
```diff
@@ -9,51 +9,139 @@ import { useNavigate } from 'react-router-dom';
...
+	const posterImage = usePosterImage( video?.guid );
+
+	const posterImagePooling = ( onGenerate = null ) => {
+		posterImage().then( ( { data: result } ) => {
+			if ( result?.generating ) {
+				setTimeout( () => posterImagePooling( onGenerate ), 2000 );
+			} else if ( result?.poster ) {
+				onGenerate?.( result?.poster );
+			}
+		} );
```
Comment (dhasilva) [inline projects/packages/videopress/src/client/admin/components/edit-video-details/use-edit-details.ts:32]: Probably needs a `.catch`.

Comment (renatoagds) [reply projects/packages/videopress/src/client/admin/components/edit-video-details/use-edit-details.ts:32]: Yep, I left a comment here: https://github.com/Automattic/jetpack/blob/2aa081d70e4b5a9034664caf639bf1380ad39660/projects/packages/videopress/src/client/admin/components/edit-video-details/use-edit-details.ts#L134 ... that we need to handle errors

**Review by dhasilva — APPROVED** [review]
LGTM :rocket: 

Screenshot_2022-09-21_15-50-58
Screenshot_2022-09-21_15-49-54

As for the reload bug, let's open another follow-up issue.

**Review by retrofox — APPROVED** [review]
LGTM. 🚀

**Pull request merged**